tidy3d/components/geometry.py

```diff
@@ -626,6 +626,36 @@ def geometry(self):
         """
         return Box(center=self.center, size=self.size)
 
+    def pop_axis(
```

Comment:
not sure I like this.. feel like it clutters things up a bit because it's for such a specific application.  How many times is the called and does it make sense to just create the normal_index = size.index(0) and feed to regular pop axis?

Reply:
It's used a few times in the frontend and a few times in the backend. It seems to me that almost every time I need to do this, I have a `plane` that I'm working with, so it seemed weird I have to specifically supply the normal index every time. I almost don't understand why this would be a geometry method if not for some simplifications it can provide for geometries that have a natural normal axis...

Reply:
Well, like you say, it's used in situations where there is a natural normal axis, like cylinders and poly slabs.  But note these are not toggled via `size`.  It's also used to compute intersections and polygons in various planes (where the plane axis is specified).  So my point is that these functions are pretty generically used and I would prefer not. hardcoding something for a special case if we dont have to.  If it were me I'd just construct and feed it the normal axis.  Or, I might try to handle it a bit differently (like define a `normal_axis` property in each Geometry that returns a value used as a default in `pop` / `unpop` axis.

Reply:
So I can partly understand your resistance to this, but firstly note that the change I made doesn't brake anything as the size selection is only triggered if axis is not supplied. There are two things that feel kinda strange to me: the first is that you have to manually define the normal axis every time even for geometries that have a natural normal axis. The second one is that it's not really clear why the method as such (staticmethod) belongs in the Geometry class (perhaps the best justification is that we don't really have hanging functions in something like a utils module). But I actually like your idea about defining a normal_axis property! This can be set to the smallest size for a Box and the extrusion direction for Cylinders and Polyslabs. Then we only need to change the Geometry pop/unpop methods to allow axis=None, and raise an error if axis == None and self.normal_aixs == None (e.g. for a sphere). What do you think?

Reply:
To your first point, it's because you dont always supply a normal axis to the pop_axis function, it could be anything, like the plot axis or whatever.  Point is that it's generic and doesn't care about the context it's being called in.

It might not belong in the Geometry class, it only does because I feel like it's better tied to some class rather than just hanging out in a utils file, and 2nd is that it's really only used for Geometry objects (things that can be plotted, intersected, etc.).  Other than that it's an arbitrary choice.

Normal axis property would work like you describe yea, but I still am a bit opposed to it unless we can define what a normal axis is and why pop_axis should have it as a default?  Is it just going to be a property defining "one of these axes is not like the other 2"?  for a specific geometry?  I'm probably being OCD but just want to think through whether we add this if it's just to avoid a few lines in other places where normal_axis is fed.  A nice thing about feeding normal axis is it's more explicit and readable.

Reply:
> Is it just going to be a property defining "one of these axes is not like the other 2"?  for a specific geometry?

Pretty much, yeah. It seems you already have something like that actually: `Planar`, where `pop_axis` is called several times with `self.axis`.

> I'm probably being OCD but just want to think through whether we add this if it's just to avoid a few lines in other places where normal_axis is fed.  A nice thing about feeding normal axis is it's more explicit and readable.

This is not a hill I want to have a long discussion on. I can revert the change. It seems that in my usage it was always tied to a plane so it felt weird to have to supply the normal every time, but I see how it's used in the plotting functions. I mean, I didn't get your point to my first point - what I was saying is that everywhere in the code up to now the axis is specifically supplied, so my change doesn't break anything. But I also see that it's not great the way it is now. So between making a normal_axis property for all geometry and just reverting the change, I'm fine with reverting.

Reply:
Ok, yea if you dont mind reverting that would make me most happy :).   But we can revisit this issue.  What makes me feel a bit icky is adding specific behavior (like checking size) to this more generic, core function that is used in a bunch of other ways.  It wouldn't break anything, but I also have been trying to avoid this out of principle in the code design (similar idea to how the components should not depend on the plugins).